public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -533,6 +535,33 @@ export function getOriginalMetricName(labelData: { [key: string]: string }) {
   return `${metricName}{${labelPart}}`;
 }
 
+function mergeHeatmapFrames(frames: DataFrame[]): DataFrame[] {
+  if (frames.length === 0) {
+    return [];
+  }
+
+  const timeField = frames[0].fields.find((field) => field.type === FieldType.time)!;
+  const countFields = frames.map((frame) => {
+    let field = frame.fields.find((field) => field.type === FieldType.number)!;
+
+    return {
+      ...field,
+      name: field.config.displayNameFromDS!,
```

Comment:
should this be the frame name?

Reply:
i think they're always the same? `displayNameFromDS` felt less arbitrary / more explicit.